tests/utils.go

```diff
@@ -1902,6 +1927,13 @@ func SkipIfNoRhelImage(virtClient kubecli.KubevirtClient) {
 	}
 }
 
+func SkipIfNoSriovDevicePlugin(virtClient kubecli.KubevirtClient) {
```

Comment:
@dankenigsberg @eedri @booxter do you think that we can organize something for CI, so that we can run sriov tests? If not we have the risk of breaking sriov on any commit without noticing.

Reply:
I did not manage to find a solution for this question just yet. I wish qemu/libvirt add support for virtual sriov, but until then - I don't know how we can possibly test it on CI. Another crazy option is to provide Lago with bare metal support, but this is not reasonable anytime soon.

Reply:
If we have a node with a sriov-device, we could pass it through to our qemu processes.

Reply:
@rmohr in theory yes, we could pass it through. I haven't got to a point where we could do it in scope of kubevirt. Are we sure this is even possible? It may be a similar case to how you can run a layered virtualization but only down to L2 but not L3. I am not an expert.

Another option, and we are looking into it, is having a separate bare metal node with the card directly deploying kubevirt without additional virtualization layer. This will require quite some work to have local based provider that deploys all components; alternatively, we build it based on kubevirt-ansible that should get support for SR-IOV anyway. Regardless, it would be a bare metal deployment which obviously has its issues (for one, it's hard or impossible to completely clean it up).

Reply:
Let's suppose we have a dedicated machine for that. We  could use the `local-provider`. Before and after each test-run we could simply destroy all containers, including the etcd data. We should then get a completely fresh cluster without much effort.

Reply:
@rmohr Yes. Stackpath folks are going to donate a machine in their lab for the cause.

We'll need to run cluster-up in a separate thread since it blocks execution. We also need to write automation for sriov components' configuration and deployment. I don't see unbearable issues with the approach.

I would like to proceed with merging the code without CI set up. We can document setup in a README file in the repo if that helps. Are you open to this approach (merging without CI set up)?

Reply:
> We'll need to run cluster-up in a separate thread since it blocks execution.

We might be able to change that.

> We also need to write automation for sriov components' configuration and deployment. I don't see unbearable issues with the approach.

Sounds great.

> 
> I would like to proceed with merging the code without CI set up. We can document setup in a README file in the repo if that helps. Are you open to this approach (merging without CI set up)?

Do we have a rough idea when we can have the testing infra to be ready? Will there be a commitment to deliver it? I would want to avoid a situation where it looks promising right now that we will get CI but then we loose focus and other things look more urgent.

And thank you for looking into it.

Reply:
I can't give exact estimates but let's just say that I am committed to it and I will push back de-prioritizing it; we also want stackpath donation to move forward irrespective of particular technical challenges with CI.

It's on my todo list. I am going to look into it in ~two weeks. (There are several other high priority SR-IOV pieces like support for multiple SR-IOV networks and kubevirt-ansible integration that I can't ignore right now.)

Also, I've just added a documentation page for SR-IOV deployment while we wait on installer and CI automation, if that helps.

Reply:
FWIW, the commitment is not only @booxter's but also mine.
Life is too short to solve the same bug twice. We need CI for SR-IOV.

Reply:
> FWIW, the commitment is not only @booxter's but also mine.
> Life is too short to solve the same bug twice. We need CI for SR-IOV.

ok, then let's go ahead.

Reply:
It would be great that in the future, we understand CI as part of the PR. It should not be an after-thought.